src/wasm.h

```diff
@@ -1365,50 +515,18 @@ class Function {
 
   Function() : result(none) {}
 
-  size_t getNumParams() {
-    return params.size();
-  }
-  size_t getNumVars() {
-    return vars.size();
-  }
-  size_t getNumLocals() {
-    return params.size() + vars.size();
-  }
+  size_t getNumParams();
+  size_t getNumVars();
+  size_t getNumLocals();
 
-  bool isParam(Index index) {
-    return index < params.size();
-  }
-  bool isVar(Index index) {
-    return index >= params.size();
-  }
+  bool isParam(Index index);
+  bool isVar(Index index);
 
-  Name getLocalName(Index index) {
-    assert(index < localNames.size() && localNames[index].is());
-    return localNames[index];
-  }
-  Name tryLocalName(Index index) {
-    if (index < localNames.size() && localNames[index].is()) {
-      return localNames[index];
-    }
-    // this is an unnamed local
-    return Name();
-  }
-  Index getLocalIndex(Name name) {
-    assert(localIndices.count(name) > 0);
-    return localIndices[name];
-  }
-  Index getVarIndexBase() {
-    return params.size();
-  }
-  WasmType getLocalType(Index index) {
-    if (isParam(index)) {
-      return params[index];
-    } else if (isVar(index)) {
-      return vars[index - getVarIndexBase()];
-    } else {
-      WASM_UNREACHABLE();
-    }
-  }
+  Name getLocalName(Index index);
+  Name tryLocalName(Index index);
```

Comment:
I mean we could rename these to `getLocalNameButIfItIsntThereReturnAnEmptyResultInstead`, but that's kind of a terrible name :P

Actually, less-jokingly we could rename the methods to `getLocalNameOrEmpty` (or `OrDefault`?).
(Do we then rename `getLocalName` to `getLocalNameOrFail`? `giveMeLocalNameOrGiveMeDeath`?)

In my head, having a descriptive name is better than having a concise comment is better than having to read the code, and that holds regardless of whether the code is simple or co-located.
One sentence of English is (usually) easier to read than one line of code, so comments can be easier to read. Comments don't have to be *correct* though.
Being able to know immediately at the call site is better than having to reference the header/implementation at all, so good naming is even better.

On the other hand, "`try*` means `get*` but with a default" is a sensible *convention*, which when followed means the calling code can be more concise. It requires the reader be familiar with the conventions, so adds a bit of mental overhead, but if it simplifies things and is used enough it can be a win.

I am almost certainly overthinking this.
I think a comment is the right change to make in this PR :P